src/components/common/modal/modals.tsx

```diff
@@ -0,0 +1,44 @@
+import { EModalType, TModalProps } from "src/context-providers";
+import { useModal } from "src/context-providers/display/modal-hook";
+import { EmptyModal } from "./empty-modal";
+import { ScrollableModal } from "./empty-modal.styles";
+
+export const Modals = () => {
+    // =========================================================================
+    // CONST, STATE, REFS
+    // =========================================================================
+    const { modals, hideModal } = useModal();
+
+    // =========================================================================
+    // HELPER FUNCTIONS
+    // =========================================================================
+    const handleBackDropClick = (type: EModalType) => {
+        hideModal(type);
+    };
+
+    const renderModalContent = (modal: TModalProps, index: number) => {
+        switch (modal.type) {
+            case EModalType.BulkEdit:
+            case EModalType.Custom:
+            case EModalType.DiscardChanges:
+                return (
+                    <ScrollableModal
+                        show={!!modal}
+                        onClick={() => handleBackDropClick(modal.type)}
+                    >
+                        <EmptyModal modal={modal} index={index} />
+                    </ScrollableModal>
+                );
+            default:
+                console.error("type given is not a modal type!");
+                break;
+        }
+    };
+
+    // =========================================================================
+    // RENDER FUNCTIONS
+    // =========================================================================
+    return (
+        <>{modals.map((modal, index) => renderModalContent(modal, index))}</>
+    );
```

Comment:
- render `ScrollableModal` here
- no need `onClick` as we dont want to dismiss via the backing
- modal is rendered according to `show` prop 
- we will render the first entry of `modals` first until there is a need to render stacked modals
- what is the `index`? why do we need to set the `key` in the `GenericModal`?

```suggestion
    return (
        <ScrollableModal show={!!modals.length}>
             {renderModalContent(modals[0])}
        </ScrollableModal>
    );
```

Reply:
fixed in the last push